src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.io.File;  // Import the File class
+import java.io.FileNotFoundException;  // Import this class to handle errors
+import java.util.ArrayList;
+import java.util.Scanner; // Import the Scanner class to read text files
+import java.io.FileWriter;
+import java.io.IOException;
+
+/**
+ * Reads and writes to a specified file path.
+ */
+public class Storage {
+    private final String FILE_PATH;
+
+    /**
+     * Create an instance to handle and writing to a specific file.
+     *
+     * @param filePath the path to the file to be read from and written to.
+     */
+    public Storage(String filePath) {
+        this.FILE_PATH = filePath;
+    }
+
+    /**
+     * Loads saved tasks from save file into an ArrayList.
+     *
+     * @return an ArrayList of tasks that are saved in the file.
+     */
+    public ArrayList<Task> load() {
+        try {
+            ArrayList<Task> savedTasks = new ArrayList<>();
+            File file = new File(FILE_PATH);
+            file.getParentFile().mkdirs();
+
+            if (file.createNewFile()) return new ArrayList<Task>();
+
+            Scanner myReader = new Scanner(file);
+            while (myReader.hasNextLine()) {
+                String data = myReader.nextLine();
+                String taskString = data.substring(8);
+                String completed = data.substring(4, 5);
+                Task newTask;
+
+                if (data.startsWith("T |")) {
+                    newTask = new Todo(taskString);
+                } else if (data.startsWith("D |")) {
+                    int dividerPosition = taskString.indexOf("|");
+                    newTask = new Deadline(taskString.substring(0, dividerPosition - 2), taskString.substring(dividerPosition + 2));
```

Comment:
Different levels of abstraction in code here. 
Easier to read if we SLAP SLAP our code :grin: ([link](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard))